### PR TITLE
Make AbstractMutableCollections satisfy expect declarations from common

### PR DESCRIPTION
**Pull request by ilya-g**
- Add actuals to AbstractMutableCollection
- Add hashCode/equals to AbstractMutableSet
- Take implementation of AbstractMutableMap from JS

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -19,9 +19,9 @@ buildKotlinVersion=1.3.0-rc-116
 buildKotlinCompilerRepo=https://teamcity.jetbrains.com/guestAuth/app/rest/builds/buildType:(id:Kotlin_130_CompilerAllPlugins),number:1.3.0-rc-116,tag:kotlin-native,pinned:true/artifacts/content/maven
 remoteRoot=konan_tests
 testDataVersion=1226829:id
-kotlinCompilerRepo=https://teamcity.jetbrains.com/guestAuth/app/rest/builds/buildType:(id:Kotlin_dev_CompilerAllPlugins),number:1.3.20-dev-1253,tag:kotlin-native,pinned:true/artifacts/content/maven
-kotlinVersion=1.3.20-dev-1253
-testKotlinVersion=1.3.20-dev-1253
+kotlinCompilerRepo=https://teamcity.jetbrains.com/guestAuth/app/rest/builds/buildType:(id:Kotlin_dev_CompilerAllPlugins),number:1.3.20-dev-1403,branch:default:any/artifacts/content/maven
+kotlinVersion=1.3.20-dev-1403
+testKotlinVersion=1.3.20-dev-1403
```
Comment (ilya-g) [inline gradle.properties]: The versions are updated to get the preview of new declarations and tests from a branch. Do not merge until it gets to master.

File: runtime/src/main/kotlin/kotlin/collections/AbstractMutableMap.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                        }
+                        return false
+                    }
+
+                    override val size: Int get() = this@AbstractMutableMap.size
+                }
+            }
+            return _keys!!
+        }
+
```
Comment (olonho) [inline runtime/src/main/kotlin/kotlin/collections/AbstractMutableMap.kt]: Two empty lines

Comment (ilya-g) [reply runtime/src/main/kotlin/kotlin/collections/AbstractMutableMap.kt]: Fixed

**Comment by olonho** [comment]
Needs rebase.

**Comment by ilya-g** [comment]
I've rebased on the latest master.
Common part is already in Kotlin master.
@vvlevchenko Please update Kotlin dependency to at least 1.3.20-dev-1601 and merge this PR.

**Comment by ilya-g** [comment]
Merged in #2310

**Pull request closed without merging**